diff --git a/commerce_google_tag_manager.module b/commerce_google_tag_manager.module
index dcb368a..eae46e4 100644
--- a/commerce_google_tag_manager.module
+++ b/commerce_google_tag_manager.module
@@ -60,7 +60,7 @@ function commerce_google_tag_manager_ecommerce_js($order) {
         $product_data = array(
           'sku' => $line_item_wrapper->commerce_product->sku->value(),
           'name' => $line_item_wrapper->commerce_product->title->value(),
-          'category' => '',
+          'category' => $line_item_wrapper->commerce_product->field_product_gtm_category->value(),
           'price' => commerce_currency_amount_to_decimal($line_item_wrapper->commerce_unit_price->amount->value(), $line_item_wrapper->commerce_unit_price->currency_code->value()),
           'quantity' => (int) $line_item_wrapper->quantity->value(),
         );
