diff --git a/editors/tinymce/linkitDialog.js b/editors/tinymce/linkitDialog.js
index 86f4726..8e34c79 100644
--- a/editors/tinymce/linkitDialog.js
+++ b/editors/tinymce/linkitDialog.js
@@ -99,4 +99,4 @@ Drupal.linkit.registerDialogHelper('tinymce', {
   }
 });
 
-})(jQuery);
\ No newline at end of file
+})(jQuery);
diff --git a/linkit.module b/linkit.module
index 78f7704..03111d1 100644
--- a/linkit.module
+++ b/linkit.module
@@ -21,6 +21,11 @@ define('LINKIT_URL_METHOD_RAW_SLASH', 2);
 define('LINKIT_URL_METHOD_ALIAS', 3);
 
 /**
+ * Define that paths should be constructed as an alias.
+ */
+define('LINKIT_URL_METHOD_ALIAS_NO_SLASH', 4);
+
+/**
  * Define which query string key Better autocomplete will just.
  */
 define('LINKIT_BAC_QUERY_KEY', 's');
@@ -1154,6 +1159,14 @@ function linkit_get_insert_plugin_processed_path(LinkitProfile $profile, $uri, $
       case LINKIT_URL_METHOD_ALIAS:
         $path = url($uri,  $options);
         break;
+      case LINKIT_URL_METHOD_ALIAS_NO_SLASH:
+        $path = url($uri,  $options);
+        if (language_default()->language != $GLOBALS["language"]->language) {
+          $path = substr($path, strlen("/".$options['language']->language));
+        }
+        $path = ltrim($path, "/");
+        break;
+
     }
 
     return $path;
diff --git a/plugins/export_ui/linkit_profiles.inc b/plugins/export_ui/linkit_profiles.inc
index e49bd07..e95054f 100644
--- a/plugins/export_ui/linkit_profiles.inc
+++ b/plugins/export_ui/linkit_profiles.inc
@@ -307,6 +307,12 @@ function _linkit_build_insert_plugin_form_fields(&$form, LinkitProfile $profile)
       '#description' => t('This will insert the alias for the result item.
         <br/>The paths will be run through the <em>url()</em> function.', array('%example_link' => 'node/123')),
     );
+    $form['data']['insert_plugin']['url_method']['#options'][LINKIT_URL_METHOD_ALIAS_NO_SLASH] = t('Alias paths, without a leading slash (/)');
+    $form['data']['insert_plugin']['url_method'][LINKIT_URL_METHOD_ALIAS_NO_SLASH] = array(
+      '#description' => t('This will strip the leading slash from alias for the result item.
+        <br/>The paths will be run through the <em>url()</em> function.'),
+    );
+
   }
 }
 
@@ -496,4 +502,4 @@ function linkit_isnumeric_validate($element, &$form_state, $form) {
   if (!empty($element['#value']) && !is_numeric($element['#value'])) {
     form_error($element, $element['#title'] . ' should only contains numbers.');
   }
-}
\ No newline at end of file
+}
